clients/lsp-csharp.el

```diff
@@ -409,24 +409,26 @@ is returned so lsp-mode can display this file."
                   (with-temp-buffer (insert-file-contents metadata-file-name)
                                     (buffer-string))))))
 
+(defun lsp-csharp--cls-download-server (_client callback error-callback update?)
+  "Install/update csharp-ls language server using `dotnet tool'.
+
+Will invoke CALLBACK or ERROR-CALLBACK based on result. Will update if UPDATE? is t"
+  (let ((exit-code (call-process "dotnet" nil nil nil
+                                 "tool"
+                                 (if update? "update" "install")
+                                 "-g" "csharp-ls")))
+    (if (zerop exit-code)
+        (funcall callback)
+      (funcall error-callback))))
```

Comment:
here you should pass a message to the error callback.  FTR we tend to use `lsp-async-start-process` to execute install processes which will pop up compilation buffer which will help the users figure out what the issue is.

Reply:
ok, I changed the code to use `lsp-async-start-process` -- seems to work